packages/shared/src/hooks/useUpdateEmailNotificationSettings.ts

```diff
@@ -0,0 +1,246 @@
+            setEmailSettings(settingsUpdate);
+            pushToast({
+              message: `Settings successfully updated. You will ${
+                settingsUpdate[settingType] ? 'no longer' : 'now'
+              } receive ${settingTitle} emails.`,
+            });
+          }
+        }
+      } catch (error) {
+        if (error instanceof Error) {
+          reportError('Failed to update email notification settings');
+        }
+        pushToast({
+          message: 'Unfortunately, there was an error updating your notification settings.',
+        });
+      }
+    },
+    [emailSettings, updateEmailNotificationSettings]
+  );
+
+  const handleToggle = useCallback(
+    async ({ settingType, settingTitle, pushToast }: handleToggleProps) => {
+      const newSettingValue = !emailSettings[settingType];
+
+      setEmailSettings((prevSettings) => ({
+        ...prevSettings,
+        [settingType]: newSettingValue,
+      }));
+
+      try {
+        await handleEmailNotificationChange({
+          settingType,
+          newSettingValue,
+          settingTitle,
+          pushToast,
+        });
+
+        pushToast({
+          message: `Settings successfully updated. You have ${
+            newSettingValue ? 'subscribed to' : 'unsubscribed from'
+          } ${settingTitle}.`,
+        });
+      } catch (error) {
+        setEmailSettings((prevSettings) => ({
+          ...prevSettings,
+          [settingType]: !newSettingValue,
+        }));
+
+        reportError('Failed to update email notification settings');
+
+        pushToast({
+          message: 'Unfortunately, there was an error updating your notification settings.',
+        });
+      }
+    },
+    [emailSettings, handleEmailNotificationChange]
+  );
+
+  return useMemo(
+    () => ({
+      emailSettings,
+      computeToggleChecked,
+      handleToggle,
+    }),
+    [emailSettings, computeToggleChecked, handleToggle]
+  );
+}
+
+type EmailNotificationSettings = {
+  notifications: boolean;
+  marketing: boolean;
+  digest: boolean;
+  membersClub: boolean;
+  all: boolean;
+  [key: string]: boolean;
+};
+
+type EmailSettings = {
+  readonly unsubscribedFromAll: boolean;
+  readonly unsubscribedFromDigest: boolean;
+  readonly unsubscribedFromMarketing: boolean;
+  readonly unsubscribedFromMembersClub: boolean;
+  readonly unsubscribedFromNotifications: boolean;
+};
+
+type updateEmailNotificationSettingsProps = {
+  unsubscribedFromNotifications: boolean;
+  unsubscribedFromDigest: boolean;
+  unsubscribedFromMarketing: boolean;
+  unsubscribedFromMembersClub: boolean;
+  unsubscribedFromAll: boolean;
+};
+
+type handleToggleProps = {
+  settingType: string;
+  settingTitle: string;
+  pushToast: ({ message }: { message: string }) => void;
+};
+
+type handleEmailNotificationChangeProps = {
+  settingType: string;
+  newSettingValue: boolean;
+  settingTitle: string;
+  pushToast: ({ message }: { message: string }) => void;
+};
```

Comment:
we can build the second type on top of the first :)

```
type handleEmailNotificationChangeProps = handleToggleProps & { newSettingsValue: boolean; }
```

Reply:
made the change!